tests/linux_kernel/helpers/test_kthread.py

```diff
@@ -15,3 +22,13 @@ def test_kthread_data(self):
             kthread_data(self.prog["drgn_test_kthread"]),
             Object(self.prog, "void *", 0xB0BA000),
         )
+
+    def test_task_is_kthread(self):
+        with fork_and_stop() as pid:
+            task = find_task(self.prog, pid)
+            is_kthread = task_is_kthread(task)
+
+            text = Path(f"/proc/{pid}/cmdline").read_text()
+            # kernel threads have an empty cmdline
+            expected = text == ""
+            self.assertEqual(is_kthread, expected)
```

Comment:
A process you create from userspace will never be a kthread. An easier and more thorough test is checking the current process (which can't be a kthread) and `init_task` (which must be a kthread):
```suggestion
        self.assertFalse(task_is_kthread(find_task(self.prog, os.getpid())))
        self.assertTrue(task_is_kthread(prog["init_task"].address_of_()))
```